lib/ansible/plugins/action/template.py

```diff
@@ -140,6 +146,9 @@ def run(self, tmp=None, task_vars=None):
             old_vars = self._templar._available_variables
             self._templar.set_available_variables(temp_vars)
             resultant = self._templar.do_template(template_data, preserve_trailing_newlines=True, escape_backslashes=False)
+            # if encode is set, we change template result encoding
+            if encode is not None:
+                resultant = resultant.encode(encode)
```

Comment:
Additionally, the use of encode can raise an error if there are characters in the template which are not present in the encoding.  We'll want to catch that exception and display a nicer error.